src/saltext/vmware/modules/vmc_networks.py

```diff
@@ -238,3 +238,277 @@ def delete(
+            Policy path to IP address pools. Maximum items it can contain is 1.
+
+        For ex:
+
+            .. code::
+
+                "advanced_config": {
+                    "address_pool_paths": [],
+                    "connectivity": "ON"
+                }
+
+    l2_extension
+        Configuration for extending Segment through L2 VPN. This field is mandatory for EXTENDED segment/network.
+        It is a json object which can contain the below fields.
+
+        'l2vpn_paths': (array of string)
+            Policy paths corresponding to the associated L2 VPN sessions
+
+        'tunnel_id': (int)
+            Tunnel ID. Minimum value is 1 and Maximum value is 4093
+
+        For ex:
+
+            .. code::
+
+                "l2_extension": {
+                    "l2vpn_paths": [
+                        "/infra/tier-0s/vmc/locale-services/default/l2vpn-services/default/sessions/c1373cd0-b2f0-11eb
+                        -80f4-d1a84667de41"
+                    ],
+                    "tunnel_id": "10"
+                }
+
+    dhcp_config_path
+        (Optional) Policy path to DHCP configuration. Policy path to DHCP server or relay configuration to
+        use for all IPv4 & IPv6 subnets configured on this segment.
+
+    display_name
+        Identifier to use when displaying entity in logs or GUI
+
+    Example values:
+
+        .. code::
+
+            {
+                "display_name":"web-tier",
+                "subnets": [
+                    {
+                        "gateway_address": "40.1.1.1/16",
+                        "dhcp_ranges": [ "40.1.2.0/24" ]
+                    }
+                ],
+                "admin_state": "UP",
+                "description": "network segment",
+                "domain_name": "net.eng.vmware.com",
+                "tags": [
+                    {
+                        "tag": "tag1",
+                        "scope": "scope1"
+                    }
+                ],
+                "advanced_config": {
+                    "address_pool_paths": [],
+                    "connectivity": "ON"
+                },
+                "l2_extension": null,
+                "dhcp_config_path": "/infra/dhcp-server-configs/default"
+            }
+
+    """
+
+    log.info("Updating Network %s for SDDC %s", network_id, sddc_id)
+    api_url_base = vmc_request.set_base_url(hostname)
+    api_url = (
+        "{base_url}vmc/reverse-proxy/api/orgs/{org_id}/sddcs/{sddc_id}/"
+        "policy/api/v1/infra/tier-1s/cgw/segments/{network_id}"
+    )
+    api_url = api_url.format(
+        base_url=api_url_base, org_id=org_id, sddc_id=sddc_id, network_id=network_id
+    )
+
+    existing_data = get_by_id(
+        hostname, refresh_key, authorization_host, org_id, sddc_id, network_id, verify_ssl, cert
+    )
+
+    # fetch the network for the given network_id
+    if vmc_constants.ERROR in existing_data:
+        return existing_data
```

Comment:
**suggestion** A test for this existing_data is missing from this PR. Not sure if there's a test covering this on `main` already... the easy way to check is to change this to 

```
if vmc_constants.ERROR in existing_data:
    assert False
    return existing_data
```

Then run `python -m nox -e tests-3 -- tests/unit/`. Currently nothing fails, so tests aren't covering this case.

Reply:
tests/unit/modules/test_vmc_networks.py has already included testcase `test_update_network_when_api_returns_no_network_with_given_id` to cover the mentioned scenario